sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -52,4 +52,22 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
 
     return await this.repo.save(newLocation);
   }
+
+  async getAllInstitutionlocations(userInfo: UserInfo): Promise<any> {
+    //To retrive institution id
+    const institutionDetails = await this.institutionService.getInstituteByUserName(
+      userInfo.userName,
+    );
+    if (!institutionDetails) {
+      throw new Error(
```

Comment:
Throw proper Http Error, here 422 un-processable. entity is more appropriate. In nest js corresponding exception class is available.

Reply:
I am not sure if the HTTP errors should be returned from the services layer. I would expect the HTTP errors to be returned from the controller layer only.